### ISSUE TITLE
Trying to build on Allwinner A20

### ISSUE THREAD

**Issue by Mathias-Fuchs** [tangrams/tangram-es#2052]
I cloned tangram-es on a board with an Allwinner A20 processor.
It runs Debian Jessie, the g++ version is 4.9.2.
Cmake runs fine, with the backport version from jessie-backports.
The g++ flags are correctly set: -std=c++14.

I still get errors that should not happen in C++14:

```
Building CXX object core/CMakeFiles/tangram-core.dir/src/map.cpp.o
In file included from /home/olimex/tangram-es/core/src/scene/drawRule.h:3:0,
                 from /home/olimex/tangram-es/core/src/style/style.h:6,
                 from /home/olimex/tangram-es/core/src/gl/mesh.h:6,
                 from /home/olimex/tangram-es/core/src/debug/textDisplay.h:4,
                 from /home/olimex/tangram-es/core/src/map.cpp:3:
/home/olimex/tangram-es/core/src/scene/styleParam.h: In constructor 'constexpr Tangram::UnitSet::UnitSet(std::initializer_lis\
t<Tangram::Unit>)':
/home/olimex/tangram-es/core/src/scene/styleParam.h:121:5: error: constexpr constructor does not have empty body
     }
     ^
In file included from /home/olimex/tangram-es/core/src/text/fontContext.h:3:0,
                 from /home/olimex/tangram-es/core/src/scene/scene.h:7,
                 from /home/olimex/tangram-es/core/src/marker/markerManager.h:4,
                 from /home/olimex/tangram-es/core/src/map.cpp:14:
/home/olimex/tangram-es/core/src/gl/glyphTexture.h: In static member function 'static constexpr Tangram::TextureOptions Tangr\
am::GlyphTexture::textureOptions()':
/home/olimex/tangram-es/core/src/gl/glyphTexture.h:12:5: error: body of constexpr function 'static constexpr Tangram::Texture\
Options Tangram::GlyphTexture::textureOptions()' not a return-statement
     }
     ^
/home/olimex/tangram-es/core/src/map.cpp: In member function 'Tangram::MapState Tangram::Map::update(float)':
/home/olimex/tangram-es/core/src/map.cpp:255:20: error: could not convert '{state}' from '<brace-enclosed initializer list>' \
to 'Tangram::MapState'
     return { state };
                    ^
core/CMakeFiles/tangram-core.dir/build.make:62: recipe for target 'core/CMakeFiles/tangram-core.dir/src/map.cpp.o' failed
make[2]: *** [core/CMakeFiles/tangram-core.dir/src/map.cpp.o] Error 1
CMakeFiles/Makefile2:311: recipe for target 'core/CMakeFiles/tangram-core.dir/all' failed
make[1]: *** [core/CMakeFiles/tangram-core.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```

Thanks for any help!

**Comment by Mathias-Fuchs** [comment]
Solved it! The C++14 support in the old g++ 4.9.2 was apparently not good enough.

**Comment by matteblair** [comment]
Thanks for the report @Mathias-Fuchs! did you end up just upgrading the GCC version?
